osu.Game.Rulesets.Osu/Objects/Drawables/DrawableHitCircle.cs

```diff
@@ -86,6 +91,13 @@ private void load()
             AccentColour.BindValueChanged(accent => ApproachCircle.Colour = accent.NewValue, true);
         }
 
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            inputManager = GetContainingInputManager();
```

Comment:
Quite a performance concern, but I guess it can be overlooked until we have pooling in?

Reply:
Indeed it is. If only the keybinding system provided the input state...

Maybe we can DI in the playfield instead, and store the current cursor position globally there?

Reply:
I think this is okay for now. LoadComplete is called on first display of each hitcircle so the overall load should be spread out enough that it doesn't matter.